src/DDTrace/Span.php

```diff
@@ -227,9 +227,15 @@ public function getAllTags()
         return $this->tags;
     }
 
+    /**
+     * @deprecated
+     * @param string $resource
+     */
     public function setResource($resource)
     {
-        $this->resource = (string)$resource;
+        error_log('DEPRECATED: Method "DDTrace\Span\setResource" will be removed soon, '
```

Comment:
I know we're using `error_log()` for the deprecated integration methods as well, but I'm curious why were using that instead of raising an `E_USER_DEPRECATED` error with `trigger_error()`.

```php
trigger_error('Method "DDTrace\Span\setResource" will be removed soon...', E_USER_DEPRECATED);
```

Sorry if I missed that discussion! :)

Reply:
This is a good point! We should refactor all the deprecation warnings like this!